### PR TITLE
185915 - documentations on all the method

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
This PR consists changes of [PBI : 185915](https://dev.azure.com/ukhydro/S-100%20Permit%20Generator/_workitems/edit/185915).

- Documentations on all the method added.

### CODE REVIEW

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

File: src/UKHO.S100PermitService.API/Controllers/PermitController.cs [hunk 1]
```diff
@@ -24,6 +24,21 @@ public PermitController(IHttpContextAccessor httpContextAccessor, ILogger<Permit
             _permitService = permitService ?? throw new ArgumentNullException(nameof(permitService));
         }
 
+        /// <summary>
```
Comment (jrippington) [inline src/UKHO.S100PermitService.API/Controllers/PermitController.cs:27]: Should just make it clear that the Permit is supplied as a Zip-encoded stream, not a file as such

File: src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs [hunk 2]
```diff
@@ -25,6 +38,19 @@ public string Decrypt(string hexString, string hexKey)
...
         }
 
+        /// <summary>
+        /// Get encrypted data.
+        /// </summary>
+        /// <remarks>
+        /// Encrypt data using secret key.
+        /// Advanced Encryption Standard (AES) block cipher algorithm is used.This is a symmetric-key algorithm. This means that the same key is used for encryption and decryption.
+        /// Cipher Block Chaining(CBC) mode of operation and No padding is used.
+        /// For S-100 size of data and secret key is fixed to 128 bits (32 characters) hexadecimal digits, if validation fails then AesEncryptionException exception handler will be trigger.
```
Comment (jrippington) [inline src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs:48]: AesEncryptionException exception will be thrown. Please update elsewhere.

File: src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs [hunk 3]
```diff
@@ -25,6 +38,19 @@ public string Decrypt(string hexString, string hexKey)
...
+        /// </remarks>
+        /// <param name="hexString">Data to be encrypt.</param>
+        /// <param name="keyHexEncoded">Secret Key.</param>
+        /// <returns>Encrypted data.</returns>
+        /// <exception cref="AesEncryptionException">AesEncryptionException exception handler triggers when length validation fails.</exception>
```
Comment (jrippington) [inline src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs:53]: Not an exception handler, that's a catch{} block. The exception is thrown.

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs [hunk 4]
```diff
@@ -48,6 +48,23 @@ public PermitService(IPermitReaderWriter permitReaderWriter,
             _permitFileConfiguration = permitFileConfiguration ?? throw new ArgumentNullException(nameof(permitFileConfiguration));
         }
 
+        /// <summary>
+        /// Get required data from dependent services and build compressed zip containing PERMIT.XML files.
```
Comment (jrippington) [inline src/UKHO.S100PermitService.Common/Services/PermitService.cs:52]: Builds a zip stream, containing Permit XML. They aren't actual files?

File: src/UKHO.S100PermitService.Common/Services/ProductKeyService.cs [hunk 5]
```diff
@@ -31,13 +31,17 @@ public ProductKeyService(ILogger<ProductKeyService> logger, IOptions<ProductKeyS
...
+        /// Get product keys from Product Key Service.
         /// </summary>
-        /// <param name="productKeyServiceRequest"></param>
-        /// /// <param name="cancellationToken"></param>
-        /// <param name="correlationId"></param>
-        /// <returns>ProductKeyServiceResponse</returns>
-        /// <exception cref="Exception"></exception>
+        /// <remarks>
+        /// If service responded with 429 TooManyRequests or 503 ServiceUnavailable StatusCodes, Then re-try mechanism will be triggered.
+        /// If service responded with other than 200 Ok StatusCode, Then PermitServiceException exception handler triggered.
```
Comment (jrippington) [inline src/UKHO.S100PermitService.Common/Services/ProductKeyService.cs:38]: Exception is thrown, not handled

**Review by jrippington — CHANGES_REQUESTED** [review]
Few things to tidy up. Permit XML is not a file, it's a stream. File implies a physical storage which isn't there. A few method descriptions refer to exception handlers being triggered, which is not correct. The exceptions are thrown (whether they are handled or not is another question for other parts of the code!)

File: src/UKHO.S100PermitService.API/Controllers/PermitController.cs (hunk 1 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.API/Controllers/PermitController.cs:27]: Resolved

File: src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs (hunk 3 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs]: Resolved

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs (hunk 4 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.Common/Services/PermitService.cs]: Resolved

File: src/UKHO.S100PermitService.Common/Services/ProductKeyService.cs (hunk 5 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.Common/Services/ProductKeyService.cs]: Resolved

File: src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs (hunk 2 above)
Comment (MastekRohitChettri) [reply src/UKHO.S100PermitService.Common/Encryption/AesEncryption.cs]: Resolved

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Pull request merged**